### PR TITLE
Add Leetcode workshop event + refactoring how events are declared

### PR DESCRIPTION
**Pull request by eth-p**

### CODE REVIEW

File: src/utils/useLayoutChecks.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import useMediaQuery from './useMediaQuery';
+import { createContext, useContext } from 'react';
+
+const context = createContext({});
+
+const queries = {
+  isDesktop: '(min-width: 600px)', // >= 600px
+  isLaptop1200: '(min-width: 1200px)', // >= than 1200px. -- FIXME(eth-p): Why is this larger than the desktop one?
+  isLaptop1392: '(min-width: 1392px)', // >= than 1392px. -- FIXME(eth-p): Why do we have two?
```
Comment (rubik516) [inline src/utils/useLayoutChecks.js:9]: Good questions in the comments, unfortunately I'm in no better position to answer these questions 😅
The `1200` and `1392` are a little bit arbitrary to me though. What is the drive behind the choice of these numbers? Do they refer to a group of devices with the same screen size...? 🤔

**Review by rubik516 — APPROVED** [review]
The changes look good to me, the website looks great when I pulled the branch. I've left a couple of questions and notes related to coding style, but those should not block merging. I've learned cool stuff from the work in this PR 😄 

Btw, in the future, can we try to group related work in separate PRs? It will be much easier and more digestible for review process 😅

**Comment by eth-p** [comment]
> Btw, in the future, can we try to group related work in separate PRs? It will be much easier and more digestible for review process 😅

Yeaaaah, that's my bad. I ended up going on a bit of a spree and adding the last two commits on a whim. I'll be better about that for next time 😅

File: src/utils/useLayoutChecks.js (hunk 1 above)
Comment (eth-p) [reply src/utils/useLayoutChecks.js:9]: I honestly have no idea. I took the values straight from the code I refactored, but I figured there's some layout-related reason for having two values (e.g. preventing unnecessary wrapping). @joshpeng1999 would be in a better position to answer that, though.

**Pull request merged**